frontend/test/metabase/scenarios/visualizations/waterfall.cy.spec.js

```diff
@@ -131,30 +131,70 @@ describe("scenarios > visualizations > waterfall", () => {
     });
   });
 
-  it("should show error for multi-series questions (metabase#15152)", () => {
-    visitQuestionAdhoc({
-      dataset_query: {
-        type: "query",
-        query: {
-          "source-table": ORDERS_ID,
-          aggregation: [["count"], ["sum", ["field-id", ORDERS.TOTAL]]],
-          breakout: [
-            ["datetime-field", ["field-id", ORDERS.CREATED_AT], "year"],
-          ],
+  describe("should show error for multi-series questions", () => {
+    it("should show errors with multiple metrics (metabase#15152)", () => {
+      visitQuestionAdhoc({
+        dataset_query: {
+          type: "query",
+          query: {
+            "source-table": ORDERS_ID,
+            aggregation: [["count"], ["sum", ["field-id", ORDERS.TOTAL]]],
+            breakout: [
+              ["datetime-field", ["field-id", ORDERS.CREATED_AT], "year"],
+            ],
+          },
+          database: SAMPLE_DB_ID,
         },
-        database: SAMPLE_DB_ID,
-      },
-      display: "line",
+        display: "line",
+      });
+
+      cy.findByText("Visualization").click();
+      cy.findByTestId("Waterfall-button").click();
+      cy.findByText("Waterfall chart does not support multiple series");
+
+      cy.findByTestId("remove-count").click();
+      cy.get(".CardVisualization svg"); // Chart renders after removing the second metric
+
+      cy.findByText("Add another series...").should("not.exist");
     });
 
-    cy.findByText("Visualization").click();
-    cy.findByTestId("Waterfall-button").click();
-    cy.findByText("Waterfall chart does not support multiple series");
+    it("should show errors with multiple dimensions", () => {
+      visitQuestionAdhoc({
+        dataset_query: {
+          type: "query",
+          query: {
+            "source-table": ORDERS_ID,
+            aggregation: [["count"]],
+            breakout: [
+              ["datetime-field", ["field-id", ORDERS.CREATED_AT], "year"],
+              ["field", PEOPLE.SOURCE, { "source-field": ORDERS.USER_ID }],
+            ],
+          },
+          database: SAMPLE_DB_ID,
+        },
+        display: "line",
+      });
+
+      cy.findByText("Visualization").click();
+      cy.findByTestId("Waterfall-button").click();
+      cy.findByText("Waterfall chart does not support multiple series");
+
+      cy.findByTestId("remove-SOURCE").click();
+      cy.findByText(
+        "Waterfall chart does not support queries with multiple groupings",
+      );
+      cy.icon("notebook").click();
 
-    cy.findByTestId("remove-count").click();
-    cy.get(".CardVisualization svg"); // Chart renders after removing the second metric
+      cy.findByText("User → Source").within(() => {
+        cy.icon("close").click();
+      });
+
+      cy.findByText("Visualize").click();
 
-    cy.findByText("Add another series...").should("not.exist");
+      cy.get(".CardVisualization svg"); // Chart renders after removing the second grouping from query
```

Comment:
I think in general we should be explicit about our assertion, I think we've discussed this in one of our FE guild meetings if I remember correctly.

```suggestion
      cy.get(".CardVisualization svg").should("exist"); // Chart renders after removing the second grouping from query
```

The code is not tested, but I found a few uses for that, so I figure it should work. e.g.
https://github.com/metabase/metabase/blob/69743e83bfa4be5f442ff99f2ed6ebd2ef882f62/frontend/test/metabase/scenarios/native/reproductions/18148-save-button-before-it-is-possible-to-save.cy.spec.js#L27